### ISSUE TITLE
Pixel project matrix not invertible error

### ISSUE THREAD

**Issue by NetisGod** [SamSamskies/react-map-gl-geocoder#28]
When I am trying to search some places(like country names), viewport does not update and I receive next error:
Uncaught Error: Pixel project matrix not invertible
    at WebMercatorViewport.Viewport (viewport.js:38)
    at new WebMercatorViewport (web-mercator-viewport.js:61)
    at fitBounds (index.js:42)
    at EventEmitter.Geocoder._this.handleResult (index.js:201)
    at EventEmitter../node_modules/events/events.js.EventEmitter.emit (events.js:81)
    at MapboxGeocoder._onChange (index.js:162)
    at Suggestions../node_modules/suggestions/src/suggestions.js.Suggestions.value (suggestions.js:174)
    at List../node_modules/suggestions/src/list.js.List.handleMouseUp (list.js:77)
    at List.<anonymous> (list.js:71)

On the other hand, when I type city name, everything is fine.
What may cause it and how to fix ?
My code is here https://github.com/NetisGod/Viewpoints-app

**Comment by SamSamskies** [comment]
Hi @NetisGod, I'm not able to reproduce this error. Could you provide a specific search query that causes this error for you? Also, please provide version numbers that you are using for this module as well as react-map-gl.

**Comment by SamSamskies** [comment]
Could you also try logging your viewport values in your render function? Let me know what the values are right before the error happens.

**Comment by NetisGod** [comment]
I type 'Kenya', for example and then I receive this error. If I type 'Kyiv' it works well. 
"react-map-gl-geocoder": "^1.7.0"
"react-map-gl": "^4.1.1"

**Comment by NetisGod** [comment]
I put console.log(this.props.viewport) in Map.jsx and received next: (Tried to find out what Geocoder writes in viewport right before crush but could not accomplish this, Geocoder crushes earlier)
https://user-images.githubusercontent.com/22257069/55030748-a7b98780-5015-11e9-9aa0-d1d2aa82c5ed.png

Sam, please, run the code at local server to find out more, here it is https://github.com/NetisGod/Viewpoints-issue

**Comment by CodyMcMichael** [comment]
I'm also seeing this issue, although the map works and go to the location. The lat long values are never updating. I'm also seeing it working with searching for a POI or feature, and failing for cities and states.

```
    "react-map-gl": "4.1.1",
    "react-map-gl-geocoder": "1.7.0",
```

**Comment by NetisGod** [comment]
It worked for me also ! Thank you ! But I can't understand how it can be an issue

**Comment by SamSamskies** [comment]
Seems like you've resolved the problem. Closing this issue. Let me know if you have any other problems.

**Comment by CodyMcMichael** [comment]
Looks like this PR could be addressing part of the issue that was referenced here. https://github.com/uber-common/viewport-mercator-project/issues/67

Removing the height and width as I stated earlier did allow me to search using the geocoder control, however it breaks my map, or forces me to set it a fixed width/height. I need to it be 100% width and height of the container.

Back to the original issue, it seems like the lat/long are never being actually passed in, possibly because of the issue I addressed above.

I can confirm this doesn't happen on addresses, but on places and countries.

Do you have any thoughts on this?

**Comment by SamSamskies** [comment]
@CodyMcMichael I haven't been able to recreate any of the issues that you're having, so it's hard to say if that PR has anything to do with your problems.

Feel free to open a new issue for each problem that you're having. Please provide a simple Code Sandbox that demonstrates your issue along with what queries cause problems for you for each issue that you open. You can fork this basic example as a starting point: https://codesandbox.io/s/x9362myl4q.

**Comment by yn5** [comment]
@CodyMcMichael you can set 100% height and with in your state object too, using strings. 

@SamSamskies I ran into the same problem and fixed it for now by putting height and width in the state object as @CodyMcMichael suggests. Tried to reproduce the issue by amending your Codesandbox starting point without any luck, tried setting all the dependency versions  to the ones I have in my project and align the code but didn't succeed in reproducing the issue in Codesandbox.

I have done a pull request https://github.com/SamSamskies/react-map-gl-geocoder/pull/38 that adds a little note about this to README to save time for future users.

**Comment by SamSamskies** [comment]
@yn5 Thanks for reporting the issue. I'll look into it more this weekend. I can probably force this error by setting the width and height of the viewport to 0 and then figure out what to do from there. I'll probably just use the default zoom instead of trying to fit zoom to the viewport in that case.

**Comment by SamSamskies** [comment]
Just pushed a new version that catches errors thrown by WebMercatorViewport when trying to zoom to bbox. Added some logging to help with debugging.

**Comment by SamSamskies** [comment]
I believe I've fixed the issue with the latest version (v2.0.5). Let me know if you still have any problems after pulling the update.

**Comment by yn5** [comment]
@SamSamskies I can confirm the issue is fixed in our app with the latest version :tada:

Thanks for your effort @SamSamskies and all other contributers to the issue!
